src/main/java/org/tools4j/hoverraft/state/CandidateState.java

```diff
@@ -116,10 +116,13 @@ private void voteForMyself(final ServerContext serverContext) {
     }
 
     private void requestVoteFromAllServers(final ServerContext serverContext, final int self) {
-        serverContext.directFactory().voteRequest()
+        final VoteRequest voteRequest = serverContext.directFactory().voteRequest()
                 .term(currentTerm())
-                .candidateId(self)
-                .sendTo(serverContext.connections().serverMulticastSender(),
+                .candidateId(self);
+
+        voteRequest.lastLogKey().copyFrom(persistentState().commandLog().lastKey());
+
+        voteRequest.sendTo(serverContext.connections().serverMulticastSender(),
```

Comment:
actually, lastLogKey() breaks the chain as it returns LogKey so I can't chain it with sendTo()